src/websocket/base.ts

```diff
@@ -239,7 +260,7 @@ abstract class BaseWebSocket extends EventEmitter {
    * Method called to send a message to the server
    */
   sendMessage(msg: string) {
-    if (!this.started) {
+    if (!this.started || !this.connected) {
```

Comment:
I feel we should add some comments at the beginning of this class explaining the difference between `this.started` and `this.connected`.

In this case, why should we set the wallet as offline if `this.connected === false`? Isn't it already marked as offline?

Reply:
Added the comments in 98ef90b9f68f919b3fe13307b3c738d12f5b4975

I'm not sure why we are setting the wallet as offline, we already do it on the `onClose` event listener, my best guess is that it was done to fix some race condition in the past, since there are no comments explaining this, I'm afraid of removing it. Do you remember anything related to this?

Reply:
Thanks for adding the comments. I feel the `this.started` has a slight different meaning. I mean, even though the websocket object might change, this flag indicates whether we should try to reconnect when the connection is gone. If this is correct, we should just fail to send the message and let the websocket reconnect.

Reply:
I have a different understanding of the `this.started` flag. Apparently, it is currently being used to prevent `this.setup` to be called when a connection is waiting to be established (until `onOpen`) is called by the event listener

Reply:
You're right! Thanks for clarifying.